include/NovelRT/Ecs/EcsUtils.h

```diff
@@ -20,11 +24,19 @@ namespace NovelRT::Ecs
      * @tparam TComponent The component type to get the ID for.
      * @return ComponentTypeId The numerical ID of the component type as an Atom.
      */
-    template<typename TComponent>[[nodiscard]] ComponentTypeId GetComponentTypeId() noexcept
+    template<typename TComponent> [[nodiscard]] ComponentTypeId GetComponentTypeId() noexcept
     {
-        static const Atom id = Atom::getNextComponentTypeId();
-        return id;
+        auto& componentTypeId = GetComponentTypeIds()[typeid(TComponent)];
+
+        if (componentTypeId == Atom(0))
+        {
+            // TODO: Two threads each trying to get a component type ID for T
```

Comment:
The simplest thing here is probably to jus do a:
```
if (componentTypeId == Atom(0))
{
    if (Interlocked.CompareExchange(&componentTypeId, expected: 0, desired: nuint.MaxValue) == 0)
    {
        componentTypeId = Atom::getNextComponentTypeId();
    } 
    else
    {
        while (Volatile.Read(&componentTypeId) == nuint.MaxValue) {
            // spin
        }
    }
}
```

In the common case, this will never be hit or the thread will have free access to update the memory
and in the uncommon case, one thread will likely do a single spin